fabulaws/library/wsgiautoscale/api.py

```diff
@@ -1890,24 +1892,30 @@ def _create_image_from_server(server):
 def _create_launch_config(type_=None, server=None):
     """Returns a new launch configuration for the specified image."""
     # Create an AMI using the deploy code, and create a new launch config.
+    created = server is None
     if server is None:
         instance_id = _create_server_for_image(type_=type_)
         # reload the environment WITH the new server
         _setup_env(env.deployment_tag, env.environment)
         server = _get_servers(env.deployment_tag, env.environment, 'web',
                               instance_ids=[instance_id])[0]
-    image = _create_image_from_server(server)
-    timestamp, changeset = image.name.split('_')[-2:]
-
-    lc = LaunchConfiguration(
-        name=_instance_name('lc', timestamp, changeset),
-        image_id=image.id,
-        security_groups=server.security_groups,
-        instance_type=_find(env.instance_types, env.environment, 'web'),
-    )
-    AutoScaleConnection().create_launch_configuration(lc)
-    print "Created a new launch config with name {0}.".format(lc.name)
-    return lc
+    try:
+        image = _create_image_from_server(server)
+        timestamp, changeset = image.name.split('_')[-2:]
+
+        lc = LaunchConfiguration(
+            name=_instance_name('lc', timestamp, changeset),
+            image_id=image.id,
+            security_groups=server.security_groups,
+            instance_type=_find(env.instance_types, env.environment, 'web'),
+        )
+        AutoScaleConnection().create_launch_configuration(lc)
+        print "Created a new launch config with name {0}.".format(lc.name)
```

Comment:
As long as we're here... maybe remove the trailing period, or add quotes around the launch config name?